flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTable.java

```diff
@@ -199,10 +208,11 @@ default int freeSlot(AllocationID allocationId) throws SlotNotFoundException {
 	/**
 	 * Return an iterator of allocated slots for the given job id.
 	 *
-	 * @param jobId for which to return the allocated slots
+	 * @param jobId for which to return the allocated slots. If {@code null} is passed,
+	 * all allocated slots held by this {@code TaskSlotTable} are returned.
 	 * @return Iterator of allocated slots.
 	 */
-	Iterator<TaskSlot<T>> getAllocatedSlots(JobID jobId);
+	Iterator<TaskSlot<T>> getAllocatedSlots(@Nullable JobID jobId);
```

Comment:
I added the annotation since the functionality of the `TaskSlotIterator` changed. I wanted to make it explicit that there is a semantic behind passing `null` now other than returning an empty iterator (which would be the expected behavior in my opinion). Or are you saying that the JavaDoc description is enough to stress on that?

Reply:
Can you clarify what you mean by "previous version". The method signature without `@Nullable`? Or are you referring to the `TaskSlotIterator` without returning all `AllocationID`s if `null` is passed?

Reply:
sorry for confusion, by "previous version", I mean where the "jobId"-methods did not have additional semantics for `jobId == null` and we had a dedicated method for that.
There I wanted to suggest having separate constructors for `TaskSlotIterator` with and without `jobId`.

Reply:
I discussed the issue again with @azagrebin offline. We came to a conclusion that I change the `TaskSlotIterator` in a way that it provides a constructor without parameters which returns all `TaskSlots`. The old constructor's functionality (taking `JobID` and `TaskSlotState`) returning no `TaskSlots` in case of passing `null` instead of a `JobID` should stay the same.